spec/controllers/hyrax/batch_ingest/batches_controller_spec.rb

```diff
@@ -48,13 +48,12 @@
     end
   end
 
-  describe 'authorization' do
-    # let(:current_user) { user }
-    let!(:admin_set) { create(:admin_set, id: 'this', with_permission_template: true) }
-    let!(:admin_set_other) { create(:admin_set, id: 'other', with_permission_template: true) }
+  describe 'authorization', clean_repo: true do
+    let(:admin_set) { create(:admin_set, id: 'this', with_permission_template: true) }
+    let(:admin_set_other) { create(:admin_set, id: 'other', with_permission_template: true) }
     let(:batch_items) { build_list(:batch_item, 1) }
     let(:batch_params) do
-      { batch: attributes_for(:batch).merge('batch_source' => fixture_file_upload('example_batches/empty.zip')) }
+      { batch: { ingest_type: 'example_ingest_type', admin_set_id: admin_set.id, submitter_email: current_user.email, batch_source: fixture_file_upload('example_batches/empty.zip') } }
     end
```

Comment:
I knew I missed the proper admin_set_id assignment here, but wasn't sure how to do it. So batch_params basically can contain any attribute of batch, and one can specify each one individually instead of using attributes_for(:batch). what is attributes_for(:batch)? does it contain the default values assigned by FactoryBot?

Reply:
The `batch_params` here is the params that the controller is accepting so putting just any attribute in there probably won't work.  I'm pretty sure that `attributes_for` is the same as `build(:batch).attributes` and it looks like our batch factory doesn't have an `admin_set_id` defined (https://github.com/samvera-labs/hyrax-batch_ingest/blob/master/spec/factories/batch.rb#L5).